compiler/noirc_frontend/src/hir/comptime/interpreter/builtin/builtin_helpers.rs

```diff
@@ -114,6 +160,47 @@ pub(crate) fn get_slice(
     }
 }
 
+/// Interpret the input as a slice, then map each element.
+/// Returns the values in the slice and the original type.
+pub(crate) fn get_slice_map<T>(
+    interner: &NodeInterner,
+    (value, location): (Value, Location),
+    f: impl Fn((Value, Location)) -> IResult<T>,
+) -> IResult<(Vec<T>, Type)> {
+    let (values, typ) = get_slice(interner, (value, location))?;
+    let values = try_vecmap(values, |value| f((value, location)))?;
+    Ok((values, typ))
+}
+
+/// Interpret the input as an array, then map each element.
+/// Returns the values in the array and the original array type.
+pub(crate) fn get_array_map<T>(
+    interner: &NodeInterner,
+    (value, location): (Value, Location),
+    f: impl Fn((Value, Location)) -> IResult<T>,
+) -> IResult<(Vec<T>, Type)> {
+    let (values, typ) = get_array(interner, (value, location))?;
+    let values = try_vecmap(values, |value| f((value, location)))?;
+    Ok((values, typ))
+}
+
+/// Get an array and convert it to a fixed size.
+/// Returns the values in the array and the original array type.
+pub(crate) fn get_fixed_array_map<T, const N: usize>(
+    interner: &NodeInterner,
+    (value, location): (Value, Location),
+    f: impl Fn((Value, Location)) -> IResult<T>,
+) -> IResult<([T; N], Type)> {
+    let (values, typ) = get_array_map(interner, (value, location), f)?;
+
+    values.try_into().map(|v| (v, typ.clone())).map_err(|_| {
+        // Assuming that `values.len()` corresponds to `typ`.
+        let Type::Array(_, ref elem) = typ else { unreachable!("it's an array") };
```

Comment:
Can we improve this panic message to something like "Expected an array type"

Reply:
Sure, I made it longer and state the that `get_array_map` is expected to have checked it was an array, which is why it's unreachable.